packages/turbo-repository/rust/src/internal.rs

```diff
@@ -100,10 +127,19 @@ impl Workspace {
                     error,
                     workspace_root: self.workspace_state.root.clone(),
                 })?;
+
         let packages = package_json_paths
             .map(|path| {
+                let pkg_json = PackageJson::load(&path)?;
                 path.parent()
-                    .map(|package_path| Package::new(&self.workspace_state.root, package_path))
+                    .map(|package_path| {
+                        let package = Package::new(
+                            pkg_json.name.unwrap(),
```

Comment:
We do not want to panic here if we find a `package.json` with no name. We shouldn't return a `package.json` without a name after [this change](https://github.com/vercel/turbo/pull/7280/files#diff-3fd939a75e56f948fbc34ce37a1ece6ef8854df4535db49183d871b14ae8ba41R286), but considering the fact that this has been in flux we should either:
 - Do the same thing and not return these package.json's without a name
 - Return an error

Reply:
updated